src/uwtools/resources/jsonschema/cdeps.jsonschema

```diff
@@ -0,0 +1,310 @@
+{
+  "$defs": {
+    "streams": {
+      "additionalProperties": false,
+      "minProperties": 1,
+      "patternProperties": {
+        "^stream0[1-9]$": {
+          "additionalProperties": false,
+          "properties": {
+            "dtlimit": {
+              "type": "number"
+            },
+            "mapalgo": {
+              "type": "string"
+            },
+            "readmode": {
+              "enum": [
+                "single"
+              ]
+            },
+            "stream_data_files": {
+              "items": {
+                "type": "string"
+              },
+              "type": "array"
+            },
+            "stream_data_variables": {
+              "items": {
+                "type": "string"
+              },
+              "type": "array"
+            },
+            "stream_lev_dimname": {
+              "type": "string"
+            },
+            "stream_mesh_file": {
+              "type": "string"
+            },
+            "stream_offset": {
+              "type": "integer"
+            },
+            "stream_vectors": {
+              "items": {
+                "type": "string"
+              },
+              "type": "array"
+            },
+            "taxmode": {
+              "type": "string"
+            },
+            "tinterpalgo": {
+              "type": "string"
+            },
+            "yearAlign": {
+              "type": "integer"
+            },
+            "yearFirst": {
+              "type": "integer"
+            },
+            "yearLast": {
+              "type": "integer"
+            }
+          },
+          "required": [
+            "dtlimit",
+            "mapalgo",
+            "readmode",
+            "stream_data_files",
+            "stream_data_variables",
+            "stream_lev_dimname",
+            "stream_mesh_file",
+            "stream_offset",
+            "stream_vectors",
+            "taxmode",
+            "tinterpalgo",
+            "yearAlign",
+            "yearFirst",
+            "yearLast"
```

Comment:
In our other schema files, we haven't put such strict requirements on the contents to be more flexible to other versions of the component that users might be working with. Do we want to take a stronger stance for CDEPS?

Reply:
Good question, which I'll leave to @uturuncoglu to answer. I'll just say that I added these with the thought that, since we will be generating these stream files from Jinja2 templates, any variables in the templates _must_ be defined in the config, or the template will not render. Of course, a user might supply a template file with more or fewer variables, in which case this list of required entries probably _should_ be flexible. They could always iterate building up their config until the template renders, but if the template contents are well-defined and very static, a strict schema could help users. I have no knowledge about this, just wanted to point out the tradeoffs.

Reply:
@christinaholtNOAA @maddenp-noaa I think having such strict requirements will not hurt. For example, certain data component like data atmosphere could get certain data mode as arguments (https://github.com/oceanmodeling/uwtools/blob/77e674de996c0b7fd2d330a34abf8a3a11574c2b/src/uwtools/resources/jsonschema/cdeps.jsonschema#L126) and it would be nice to impose it since user could provide a non existing data mode as argument and the cdeps will fail. Of course, this also requires updating the scheme file once new data mode is implemented in CDEPS side and want to use with the UWtools. So, there is no any right way to go but just we need to pick a once. At this point, we are using a more conservative way but maybe we could adjust it in the future.

Reply:
I am okay with this. If we need to take a different route in the future, we have a few options to make it more flexible. I think those possibilities are making this level of strictness a bit more tolerable as a first pass. 

Thanks for your contributions and clarifications!